### PR TITLE
Cleaning up ProductPage mess

### PR DESCRIPTION
**Pull request by fromITzerotoIThero**

### CODE REVIEW

File: src/pages/ProductPage/ProductPage.jsx [hunk 1]
```diff
@@ -2,14 +2,17 @@ import React from 'react';
...
 import styles from './ProductPage.module.css';
+import ShoppingCartOutlinedIcon from '@mui/icons-material/ShoppingCartOutlined';
 
 function ProductPage() {
 
     const { id } = useParams();
     const [product, setProduct] = useState(null);
 
+    const isDiscounted = false;
+    const currentPrice = null;
```
Comment (piotrwicijowski) [inline src/pages/ProductPage/ProductPage.jsx:14]: Are those values used?

File: src/pages/ProductPage/ProductPage.jsx [hunk 2]
```diff
@@ -20,20 +23,43 @@ function ProductPage() {
...
+                                }
+                            </div>
+                            <button className={styles.cart_btn}>
+                                <span>Add to cart</span>
+                                <ShoppingCartOutlinedIcon className={styles.cart_icon} />
+                            </button>
+                        </div>
+                    </div>
+                    <div>
+                        <img src={product.thumbnail_urls} alt={product.name} />
```
Comment (piotrwicijowski) [inline src/pages/ProductPage/ProductPage.jsx:57]: `thumbnail_urls` is an array, at least for now get the first element from it (assume that it always will have at least one)

File: src/pages/ProductPage/ProductPage.jsx [hunk 3]
```diff
@@ -20,20 +23,43 @@ function ProductPage() {
             setProduct(data[0]);
         }
         fetchData();
-    }, []);
+    }, [id]);
 
 
     return (
         <div>
```
Comment (piotrwicijowski) [inline src/pages/ProductPage/ProductPage.jsx:30]: Do we need the wrapping div?

File: src/pages/ProductPage/ProductPage.module.css [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+.price_row {
+    font-size: 24px;
+    padding: 4px;
+}
+
+.cart_btn {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    min-width: 9.5em;
```
Comment (piotrwicijowski) [inline src/pages/ProductPage/ProductPage.module.css:29]: Do we need the min width? Shouldn't the size be determined by the content and the padding?

File: src/pages/ProductPage/ProductPage.module.css [hunk 5]
```diff
@@ -1,3 +1,40 @@
...
+    padding: 4px 8px;
+    background-color: var(--default-color);
+    color: #fff;
+    border: none;
+    font-size: 0.75rem;
+}
+
+.description {
+    padding: 2rem 2rem 2rem 1rem;
```
Comment (piotrwicijowski) [inline src/pages/ProductPage/ProductPage.module.css:38]: Why not just `padding: 2rem` all the way?

**Review by piotrwicijowski — CHANGES_REQUESTED** [review]
Waiting for your changes

File: src/pages/ProductPage/ProductPage.jsx (hunk 1 above)
Comment (fromITzerotoIThero) [reply src/pages/ProductPage/ProductPage.jsx:14]: Waiting for them to be added to the database.

File: src/pages/ProductPage/ProductPage.jsx (hunk 3 above)
Comment (fromITzerotoIThero) [reply src/pages/ProductPage/ProductPage.jsx:30]: Yeah, we do need it!

**Review by piotrwicijowski — APPROVED** [review]

**Comment by fromITzerotoIThero** [comment]
So the PR shows as a yet another task! How cool is that! 
Seems to be a good place to ask more questions about the VS-Git-GitHub workflow.

1. I've noticed a button `Publish Branch` in Source Control - do I need to do it before I commit to the branch or does that happen automatically? 
2. Say I work on a feature but I know it's far from a PR (because I do a PR when I think that the code complete and it only waits for your review and my post-review changes, right?) what is the best way to ask for a quick code review? Make a commit on the feature branch and ask you to have a look?

**Comment by piotrwicijowski** [comment]
1. If you create a branch locally and try to push to that, then VS Code will ask you to publish it, so it does the `Publish Branch` for you behind the scenes.
2. You can create a "draft PR" for that - it is signalling "I'm not done", but at the same time it gives the reviewer a nice place to have a conversation about the changes that you are working on. In order to do so, start creating a pull request, and on the "create PR" button there is a dropdown on the right which allows you to create a draft PR instead. I highly recommend it.

**Comment by fromITzerotoIThero** [comment]
Oh that sounds just great, thx!!! 
I have another question - I wanted to check out for another branch so I clicked the branch icon in the bottom left corner, had the Command Palette open up and what I saw where branches with 'the tree' icon (local, I assume) and branches with 'the cloud' called `origin` (remote) - when do I use the latter?

IMG-9436

**Comment by piotrwicijowski** [comment]
99% of the time the local branch state will match the remote branch state, so choosing one over the other should make no difference, so just go with the "tree icon", which indicates local branch. Also - always make sure to do a Pull, this would make sure that the latest changes from github are reflected in your machine, and that your local branch is matching with remote branch.
The only time when it makes sense to check out the remote branch, is when looking at someone else's code - then the branch exists only on remote, but in order to run it on your machine you need to have a local corresponding branch. I hope it makes sense.

**Pull request merged**